app/src/main/java/org/simple/clinic/scanid/ScanSimpleIdScreenController.kt

```diff
@@ -39,21 +43,40 @@ class ScanSimpleIdScreenController @Inject constructor(
         .map { scannedQrCode ->
           try {
             val bpPassportCode = UUID.fromString(scannedQrCode)
-            ScanSimpleIdScreenPassportCodeScanned.ValidPassportCode(bpPassportCode)
+            ValidPassportCode(bpPassportCode)
           } catch (e: IllegalArgumentException) {
-            ScanSimpleIdScreenPassportCodeScanned.InvalidPassportCode
+            InvalidPassportCode
           }
         }
 
     events.mergeWith(scannedBpPassportCodes)
   }
 
   private fun handleScannedBpPassportCodes(events: Observable<UiEvent>): Observable<UiChange> {
-    val scannedBpPassportCodeStream = events
-        .ofType<ScanSimpleIdScreenPassportCodeScanned.ValidPassportCode>()
+    val showKeyboardEvents = events
```

Comment:
I am not entirely sure of this because it seems like the controller is too deeply coupled with the implementation of the screen and the keyboard behaviour. I wonder if we should consider delegating the behaviour of filtering the scanned passport codes while the keyboard is up to the view layer and make the controller not have to care about this.

Using the [RxJavaExtensions#valve](https://github.com/akarnokd/RxJavaExtensions#flowabletransformersvalve) operator might actually be really helpful here as well.

Reply:
We did this because of the way how keyboard scanning is implemented in the controller now. It is just exposed as a stream of scan events. And the controller does not have access to a camera / QR code scanner interface.

Reply:
I prefer the `valve` operator, however that would require a refactor which we may not be able to pull off in this iteration.

Reply:
Sure, let's revisit this after the feature is complete.